moninag/tests/unittests/contact/test_views.py

```diff
@@ -112,34 +112,158 @@ def test_get_not_existing_contact(self):
+        expected_json_response['error'] = 'User with such email already exists.'
+
+        self.assertEqual(response.status_code, 400)
+        self.assertJSONEqual(response.content.decode('utf-8'), json.dumps(expected_json_response))
+
+    def test_post_invalid_email(self):
+        """Ensure that POST fails to create contact with invalid email and returns 400 status"""
+
+        url = reverse('contacts')
+        data = json.dumps({'first_name': 'test_first_name',
+                           'second_name': 'test_second_name',
+                           'email': 'invalid-email.com'})
+
+        response = self.client.post(url, data=data, content_type='application/json')
+
+        expected_json_response = {}
+        expected_json_response['error'] = 'Invalid email format.'
+
+        self.assertEqual(response.status_code, 400)
+        self.assertJSONEqual(response.content.decode('utf-8'), json.dumps(expected_json_response))
+
+    def test_verify(self):
+        """Ensure that GET method returns expected url and 200 status."""
+
+        contact = Contact.objects.get(id=101)
+        contact.is_active = True
+        contact.save()
+
+        url = reverse('verify', args=[contact.activation_key])
+        response = self.client.get(url)
+
+        expected = 'contact/verified.html'
+
+        self.assertEqual(response.status_code, 200)
+        self.assertTemplateUsed(response, expected)
+
     def test_put(self):
         """Ensure that PUT method returns fully updated contact and 200 status."""
 
-        url = reverse('contact', args=[1])
+        url = reverse('contact', args=[102])
         data = json.dumps({'first_name': 'changedFirstName',
                            'second_name': 'changedSecondName',
                            'email': 'changed@email.com'})
 
         response = self.client.put(url, data=data, content_type='application/json')
-        contact = Contact.objects.get(id=1)
+        contact = Contact.objects.get(id=102)
 
         self.assertEqual(response.status_code, 200)
         self.assertEqual(contact.first_name, 'changedFirstName')
         self.assertEqual(contact.second_name, 'changedSecondName')
         self.assertEqual(contact.email, 'changed@email.com')
 
-    def test_put_incorrect_format(self):
-        """Ensure that PUT method returns 400 status."""
+        expected_json_response = {}
+        expected_json_response['response'] = contact.to_dict()
 
-        url = reverse('contact', args=[1])
+        self.assertEqual(response.status_code, 200)
+        self.assertJSONEqual(response.content.decode('utf-8'), json.dumps(expected_json_response))
+
+    def test_put_invalid_field(self):
+        """Ensure that PUT method returns error and 400 status."""
+
+        url = reverse('contact', args=[101])
         data = json.dumps({'first_name': 'newFirstName',
-                           'second_nme': 'newSecondName'})
+                           'secon_name': 'newSecondName'})
```

Comment:
@s0812 It would be cleaner if name it  `invalid_field`, but it is optional :slightly_smiling_face: